CMakeLists.txt

```
@@ -27,7 +27,7 @@ if (SCN_IS_TOP_PROJECT AND NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release CACHE STRING "")
 endif()
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
```

```suggestion
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
```

I think you'll want paths relative to `CMAKE_CURRENT_SOURCE_DIR` in nearly all cases. And paths relative to `CMAKE_CURRENT_BINARY_DIR` in almost all of the remaining cases.

---

I'm less confident in replacing `PROJECT_BINARY_DIR` below, because it needs to match the rest of the CMakeLists.txt in scnlib. I'd recommend cleaning that up too, though I haven't tried myself... possibly it's a bigger refactor for another PR.

---

I agree it probably would make sense to change the two places I added `PROJECT_BINARY_DIR` to use `CMAKE_CURRENT_BINARY_DIR` instead but that would change the layout of the binary tree and I wasn't sure if that would break some existing use.